airbyte-integrations/connectors/source-square/source_square/spec.json

```diff
@@ -4,33 +4,99 @@
-        "default": true
+        "default": false
       },
       "start_date": {
         "type": "string",
         "description": "The start date to sync data. Leave blank for full sync. Format: YYYY-MM-DD.",
         "examples": ["2021-01-01"],
-        "default": "1970-01-01",
+        "default": "2021-01-01",
         "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}$"
       },
       "include_deleted_objects": {
         "type": "boolean",
         "description": "In some streams there is and option to include deleted objects (Items, Categories, Discounts, Taxes)",
         "examples": [true, false],
         "default": false
+      },
+      "credentials": {
+        "type": "object",
+        "title": "Credential Type",
+        "oneOf": [
+          {
+            "title": "Oauth authentication",
+            "type": "object",
+            "required": [
+              "auth_type",
+              "client_id",
+              "client_secret",
+              "refresh_token"
+            ],
+            "properties": {
+              "auth_type": {
+                "type": "string",
+                "const": "Oauth",
+                "enum": ["Oauth"],
+                "default": "Oauth",
+                "order": 0
+              },
+              "client_id": {
+                "title": "Client ID",
+                "type": "string",
+                "description": "The Square-issued ID of your application",
+                "airbyte_secret": true
+              },
+              "client_secret": {
+                "title": "Client Secret",
+                "type": "string",
+                "description": "The Square-issued application secret for your application",
+                "airbyte_secret": true
+              },
+              "refresh_token": {
+                "title": "Refresh Token",
+                "type": "string",
+                "description": "A refresh token generated using the above client ID and secret",
+                "airbyte_secret": true
+              }
+            }
+          },
+          {
+            "type": "object",
+            "title": "API Key",
+            "required": ["auth_type", "api_key"],
+            "properties": {
+              "auth_type": {
+                "type": "string",
+                "const": "Apikey",
+                "enum": ["Apikey"],
+                "default": "Apikey",
+                "order": 1
+              },
+              "api_key": {
+                "title": "API key token",
+                "type": "string",
+                "description": "The API key for a Square application",
+                "airbyte_secret": true
+              }
+            }
+          }
+        ]
       }
     }
+  },
+  "authSpecification": {
+    "auth_type": "oauth2.0",
+    "oauth2Specification": {
+      "rootObject": ["credentials", 0],
+      "oauthFlowInitParameters": [["client_id"], ["client_secret"]],
+      "oauthFlowOutputParameters": [["refresh_token"]]
+    }
```

Comment:
Could you look at spec.json from `source-harvest` as a reference and start adding the `advanced_auth` block in the spec.json?

We'll be deprecating the `authSpecification` part and will need the new `advanced_auth` instead.

Thanks